### PR TITLE
Add Kernel and hardware abstraction

### PR DESCRIPTION
**Pull request by doniks**

### CODE REVIEW

**Comment by doniks** [comment]
not finished yet

**Comment by doniks** [comment]
not finished yet. browse CI build here: https://ci.ubports.com/job/docs.ubports.com/job/PR-306/1/artifact/_build/html/index.html

File: systemdev/kernel-hal.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+We distinguish a few different groups of Ubuntu Touch devices with respect to how the kernel and hardware abstraction is implemented:
+
+  * Android 5.1 based ports
+  * Halium based ports
+  * Linux kernel
+
+Android 5.1 based ports
+-----------------------
+
+Android 5.1 based ports of Ubuntu Touch consist of the Linux kernel for this device plus a minimal Android system that is used to enable all the hardware. Every device has it's own fork of the Linux kernel and they all quite old and heavily modified for the specific device by the manufacturer of that device.
```
Comment (talmai) [inline systemdev/kernel-hal.rst:23]: Consider: "Every device has it's own fork on the Linux kernel and they are all heavily modified for the specific hardware specification by the manufacturer of that device"

File: systemdev/kernel-hal.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+For some devices there is continuous integration (CI) set up to build the Kernel and the Android system. Other devices have to be built manually from their repositories. For a third group of devices, we do not have the source code available. The manufacturers of these devices provided the prebuild binaries.
```
Comment (talmai) [inline systemdev/kernel-hal.rst:25]: Consider: "For some devices continuous integration (CI) has been set up to build both the Kernel as well as the Android system. Other devices have to be built manually from their repositories. Unfortunately, there are still devices for which we do not have the source code available. The manufacturers of these devices provided the prebuild binaries, instead."

File: systemdev/kernel-hal.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    * Meizu MX4 (arale)
+    * Meizu Pro 5 (turbo)
+    * Nexus 7 2013 GSM (deb) - Note: The deb build was not done by the manufacturer but by a community member. Unfortunately those repositories have disappeared, so only the prebuilt binaries remain available.
+
+Further details can be found under  :doc:`Legacy porting <ubp-5.1>` and in the https://gitlab.com/ubports/community-ports
+
+Halium based ports
+------------------
+
+Newer ports to Android devices are based on Halium. In some aspects Halium is similar to the 5.1 based ports. It also uses the device specific fork of the Linux kernel provided by the manufacturer. And it also uses a minimal Android system to enable some hardware. Halium is different from the 5.1 based ports by being a more generic way of porting to Android devices. This allows the work to be shared between multiple projects that bring differnt flavours of Linux systems to Android devices. Halium uses newer versions of Android.
```
Comment (talmai) [inline systemdev/kernel-hal.rst:48]: Consider: 

"Newer ports to Android devices are based on Halium. In some aspects Halium is similar to the 5.1 based ports: 

* It also uses the device specific fork of the Linux kernel provided by the manufacturer. 
* It also uses a minimal Android system to enable some hardware. 

However, Halium permits a more generic way of porting to Android devices. This allows the work to be shared between multiple projects that bring different flavours of Linux systems to Android devices. This is what we have started to see on newer versions of Android."

File: systemdev/kernel-hal.rst (hunk 1 above)
Comment (doniks) [reply systemdev/kernel-hal.rst]: pls recheck

File: systemdev/kernel-hal.rst (hunk 2 above)
Comment (doniks) [reply systemdev/kernel-hal.rst]: pls recheck

File: systemdev/kernel-hal.rst (hunk 3 above)
Comment (doniks) [reply systemdev/kernel-hal.rst]: pls recheck

File: systemdev/kernel-hal.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+   * Meizu MX4 (arale)
+   * Meizu Pro 5 (turbo)
+   * Nexus 7 2013 GSM (deb) - Note: The deb build was not done by the manufacturer but by a community member. Unfortunately those repositories have disappeared, so only the prebuilt binaries remain available.
+
+Detailed steps for the three CI enabled devices can be found under  :doc:`HAL for Nexus 5, OnePlus One, and Fairphone 2 <ubp-5.1>`.
+
+For the Nexus 7 2013 (flo) head  over to the `flo community-ports repository<https://gitlab.com/ubports/community-ports/flo>`.
+
+TODO: For the Nexus 4 (make) .... dunno you tell me! binaries: https://gitlab.com/ubports/community-ports/mako
+
```
Comment (doniks) [inline systemdev/kernel-hal.rst:48]: @UniversalSuperBox any relevant info we can put here for N4?

File: systemdev/kernel-hal.rst [hunk 5]
```diff
@@ -20,9 +20,9 @@ We distinguish a few different groups of Ubuntu Touch devices with respect to ho
 Android 5.1 based ports
 -----------------------
 
-Android 5.1 based ports of Ubuntu Touch consist of the Linux kernel for this device plus a minimal Android system that is used to enable all the hardware. Every device has it's own fork of the Linux kernel and they all quite old and heavily modified for the specific device by the manufacturer of that device.
+Android 5.1 based ports of Ubuntu Touch consist of the Linux kernel for this device plus a minimal Android system that is used to enable all the hardware. Every device has it's own fork of the Linux kernel. They are all heavily modified for the specific hardware used in that device. These forks are also based of quite old Linux kernel versions.
```
Comment (talmai) [inline systemdev/kernel-hal.rst:23]: ...based 'off' of quite old...

File: systemdev/kernel-hal.rst [hunk 6]
```diff
@@ -20,9 +20,9 @@ We distinguish a few different groups of Ubuntu Touch devices with respect to ho
...
 
-For some devices there is continuous integration (CI) set up to build the Kernel and the Android system. Other devices have to be built manually from their repositories. For a third group of devices, we do not have the source code available. The manufacturers of these devices provided the prebuild binaries.
+For some devices continuous integration (CI) has been set up to build both the Kernel as well as the Android system. Other devices have to be built manually from their repositories. For a third group of devices, we do unfortunately, not have the source code available. The manufacturers of these devices provided the binaries builds instead.
```
Comment (talmai) [inline systemdev/kernel-hal.rst:25]: ...we do not unfortunately, have the source code...

File: systemdev/kernel-hal.rst [hunk 7]
```diff
@@ -24,46 +24,54 @@ Android 5.1 based ports of Ubuntu Touch consist of the Linux kernel for this dev
...
+* Prebuilt binaries
+   * Bq Aquaris E4.5 (krillin)
+   * Bq Aquaris E5 (vegetahd)
+   * Bq Aquaris M10 HD (cooler)
+   * Bq Aquaris M10 FHD (frieza)
+   * Meizu MX4 (arale)
+   * Meizu Pro 5 (turbo)
+   * Nexus 7 2013 GSM (deb) - Note: The deb build was not done by the manufacturer but by a community member. Unfortunately those repositories have disappeared, so only the prebuilt binaries remain available.
+
+Detailed steps for the three CI enabled devices can be found under  :doc:`HAL for Nexus 5, OnePlus One, and Fairphone 2 <ubp-5.1>`.
```
Comment (talmai) [inline systemdev/kernel-hal.rst:43]: :+1:

File: systemdev/ubp-5.1.rst [hunk 8]
```diff
@@ -1,10 +1,12 @@
-Legacy porting
-==============
+hammerhead, bacon and FP2
```
Comment (talmai) [inline systemdev/ubp-5.1.rst:1]: This is obvious, but not so for someone who is starting from scratch.. I'd suggest being more explicit here:

"Nexus 5 (hammerhead), OnePlus One (bacon) and Fairphone 2 (FP2)"

File: systemdev/ubp-5.1.rst [hunk 9]
```diff
@@ -1,10 +1,12 @@
...
+=========================
 
-TOOD: fix the title
-TODO: can we turn this into a build instruction for ALL 5.1 devs? e.g., what's the difference btw Opo and N7?
 
-This page documents the resources and process to build an Android compatibility image for the LG Nexus 5, OnePlus One, or Fairphone 2. This information should not be used to bring up any new devices, only to update these older images. All new ports should be created using Halium following the documentation starting at :doc:`introduction <../porting/introduction>`. See also the general overview for :doc:`kernel-hal`.
+This page documents how to build the Kernel and the Android image for the LG Nexus 5 (hammerhead), OnePlus One (bacon), or Fairphone 2 (FP2).
```
Comment (talmai) [inline systemdev/ubp-5.1.rst:5]: And if you choose to implement the previous comment, we can either leave it as is, or here use just the codenames. Your choice.

File: systemdev/ubp-5.1.rst [hunk 10]
```diff
@@ -1,10 +1,12 @@
...
+
+.. note::
+
+    This information should not be used to bring up any new devices, only to update these older images. All new ports should be created using Halium following the documentation starting at :doc:`introduction <../porting/introduction>`.
```
Comment (talmai) [inline systemdev/ubp-5.1.rst:9]: What about...

"We highly suggest that you only use non-halium builds for these older devices, as you may find yourself with no support. All new ports should be created...."

File: systemdev/ubp-5.1.rst (hunk 8 above)
Comment (doniks) [reply systemdev/ubp-5.1.rst:1]: Yeah ... the problem is that the title of the page is also used in the navigation on the left side ... and then it gets really long and looks ugly there .... not sure how to best solve this

**Comment by doniks** [comment]
ok. ready as far as I'm concerned. Please review

**Comment by doniks** [comment]
hold on. I just learned that some more kernel sources are available https://forums.ubports.com/topic/4509/otg-ethernet/6?_=1593114686119 I'll add that in the next days

**Comment by doniks** [comment]
fixed. ready for review (again :-P )

**Pull request merged**